### ISSUE TITLE
RDSInstance: missing masterUsername & password in secret

### ISSUE THREAD

**Issue by OferPRTZ** [crossplane/provider-aws#776]
What happened?

When applying RDSInstance according to the documentation seems that masterUsername & password are missing in the exported secret.
applied config:

```apiVersion: database.aws.crossplane.io/v1beta1
kind: RDSInstance
metadata:
  name: testpostgres
spec:
  deletionPolicy: Delete
  forProvider:
    dbInstanceClass: db.t2.small
    storageEncrypted: true
    masterUsername: root
    enableCloudwatchLogsExports: ["postgresql", "upgrade"]
    vpcSecurityGroupIDRefs:
      - name: vpcname
    dbSubnetGroupNameRef:
      name: subnetgroup
    allocatedStorage: 20
    engine: postgres
    engineVersion: "13.2"
    region: us-gov-east-1
    multiAZ: false
    backupRetentionPeriod: 1
    skipFinalSnapshotBeforeDeletion: true
    preferredBackupWindow: 22:30-23:00
    preferredMaintenanceWindow: sat:23:32-sun:00:02
  providerConfigRef:
    name: aws-provider
  providerRef:
    name: aws-provider
  writeConnectionSecretToRef:
    name: rds-test-secret
    namespace: rds-test
```

The created secret doesn't contain the masterUsername nor password as instead expected:

```
Name:      rds-test-secret   
Namespace:    rds-test
Labels:       <none>
Annotations:  <none>

Type:  connection.crossplane.io/v1alpha1

Data
====
endpoint:  78 bytes
port:      4 bytes

```
What environment did it happen in?

Crossplane version: v1.2.2
crossplane/provider-aws-controller: v0.19.0

**Comment by gjacobsatl** [comment]
When this happens to me it's because I already have an instance with that same name running in RDS (either due to Orphan deletionPolicy or a previous failed deletion).

**Comment by mefistofeel** [comment]
Faced with the same issue. 
What is interesting, 1 time from 20 I have fulfilled credentials in a Secret. In other times I see only `endpoint` and `port`
My env:
Openshift 4.7
Crossplane : 1.3.0
AWS-provider: 0.19.0

**Comment by haarchri** [comment]
Fixed with #1197
